7alip_week2/more-js/app.js

```diff
@@ -0,0 +1,207 @@
+  console.log(`a ${color} car`);
+}
+
+// 3. Create an object and a function that takes the object as a parameter and prints out all of its properties and values.
+function showObj(obj) {
+  let objKeys = Object.keys(obj);
+
+  let objVals = Object.values(obj);
+
+  for (let i = 0; i < objKeys.length; i++) {
+    console.log(`The value of the property '${objKeys[i]}' is '${objVals[i]}'`)
+  }
+}
+
+const myPerson = { name: 'Ali', age: 10 };
+
+showObj(myPerson); // The value of the property 'name' is 'Ali' // The value of the property 'age' is '10'
+
+
+// 4. Create a function named vehicleType that receives a color, and a code, 1 for car, 2 for motorbike.And prints 'a blue motorbike' for example when called as vehicleType("blue", 2)
+
+function vehicleType(color, code) {
+  if (code === 1) {
+    console.log(`a ${color} car`);
+  } else if (code === 2) {
+    console.log(`a ${color} motorbike`);
+  } else {
+    console.log("Invalid code!");
+  }
+}
+
+// 5. Can you write the following without the if statement, but with just as a single line with console.log(...);?
+/*
+if (3 === 3) {
+  console.log("yes");
+} else {
+  console.log("no");
+}
+*/
+
+3 === 3 ? console.log("yes") : console.log("no");
+
+
+// 6. Create a function called vehicle, like before, but takes another parameter called age, so that vehicle("blue", 1, 5) prints 'a blue used car'
+
+function vehicle(color, code, age) {
+  let status = age === 0 ? "new" : "used";
+
+  switch (code) {
+    case 1:
+      console.log(`${color} ${status} car`)
+      break;
+    case 2:
+      console.log(`${color} ${status} motorbike`)
+      break;
+    default:
+      console.log("Invalid code");
+      break;
+  }
+}
+
+// 7. Make a list of vehicles, you can add "motorbike", "caravan", "bike", or more.
+
+const vehicles = ["motorbike", "caravan", "car", "truck"];
+
+// 8. How do you get the third element from that list ?
+
+let thirdItem = vehicles[2];
+console.log(thirdItem); // car
+
+
+// 9. Change the function vehicle to use the list of question 7. So that vehicle("green", 3, 1) prints "a green new bike".
+
+// Solution is in 6.
+
+// 10. Use the list of vehicles to write an advertisement.So that it prints something like: "Amazing Joe's Garage, we service cars, motorbikes, caravans and bikes.". (Hint: use a for loop.)
+
+// Hint, the output should be correct English with all the punctuation in place(that's the challenge). So plurals for the vehicle types, commas followed by a single space, the word and to replace the final comma and closed off by a period.
+
+let adStr = "Amazing Joe's Garage, we service ";
+
+function ad(str, arr) {
+  for (let i = 0; i < arr.length; i++) {
+    const currentEl = arr[i];
+    const lastEl = arr[arr.length - 1];
+
+    if (currentEl !== lastEl) {
+      str += `${currentEl}s, `;
+    } else {
+      str += `${lastEl}s.`;
+    }
+  }
+
+  return str;
+}
+
+console.log(ad(adStr, vehicles)); // Amazing Joe's Garage, we service motorbike, caravan, car, truck.
```

Comment:
According to the instruction it should be ` Amazing Joe's Garage, we service motorbike, caravan, car and truck.`, So you just need to add `and` before the last word. Overall good job!